libminifi/test/resources/TestOnScheduleRetry.yml

```diff
@@ -0,0 +1,40 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+Flow Controller:
+    name: MiNiFi Flow
+    id: 2438e3c8-015a-1000-79ca-83af40ec1990
+    onschedule retry interval: 1000 ms
+Processors:
+    - name: kamikaze
+      id: 2438e3c8-015a-1000-79ca-83af40ec1991
+      class: org.apache.nifi.processors.KamikazeProcessor
+      max concurrent tasks: 1
+      scheduling strategy: TIMER_DRIVEN
+      scheduling period: 100 sec
+      penalization period: 30 sec
+      yield period: 1 sec
+      run duration nanos: 0
+      auto-terminated relationships list:
+      Properties:
+        Throw in onSchedule: true
```

Comment:
This is a good test, but I would like to have other tests that test:
  - recovery (onSchedule throwing, then not throwing on retry, meaning onSchedule is not called anymore, onTrigger is called on the other hand)
 - onTrigger throwing and proper logs are created about the rollback (now that this sets it to `warn` instead of `debug`

Reply:
Done, the integration test was extended with recovery and onTrigger calls as well.